File: app/src/main/java/eventcoordinator2017/myevent/ui/main/MainPageFragmentAdapter.java
package eventcoordinator2017.myevent.ui.main;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import java.util.List;

import eventcoordinator2017.myevent.model.data.Event;
import eventcoordinator2017.myevent.ui.main.tabs.MainPageFragment;

/**
 * Created by Mark Jansen Calderon on 2/8/2017.
 */

public class MainPageFragmentAdapter extends FragmentPagerAdapter {
    private Context context;
    private List<String> strings;

    public MainPageFragmentAdapter(FragmentManager fm, Context context, List<String> strings) {
        super(fm);
        this.context = context;
        this.strings = strings;
    }

    @Override
    public int getCount() {
        return strings.size();
    }

    @Override
    public Fragment getItem(int position) {
        return MainPageFragment.newInstance(position + 1, strings.get(position));
    }

    @Override
    public CharSequence getPageTitle(int position) {
        // Generate title based on item position
        return strings.get(position);
    }
}

File: app/src/main/java/eventcoordinator2017/myevent/ui/main/MainPageFragmentAdapter.kt
package eventcoordinator2017.myevent.ui.main

import android.content.Context
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

import eventcoordinator2017.myevent.ui.main.tabs.MainPageFragment

/**
 * Created by Mark Jansen Calderon on 2/8/2017.
 */

class MainPageFragmentAdapter(fm: FragmentManager, private val context: Context, private val strings: List<String>) : FragmentPagerAdapter(fm) {

    override fun getCount(): Int {
        return strings.size
    }

    override fun getItem(position: Int): Fragment {
        return MainPageFragment.newInstance(position + 1, strings[position])
    }

    override fun getPageTitle(position: Int): CharSequence? {
        // Generate title based on item position
        return strings[position]
    }
}